### PR TITLE
[LPT] CNNNetwork LPT removal

### PR DESCRIPTION
**Pull request by eshoguli**

### CODE REVIEW

File: inference-engine/tests_deprecated/functional/cldnn/CMakeLists.txt [hunk 1]
```diff
@@ -17,15 +17,6 @@ file(GLOB CLDNN_TEST_SOURCES
         ${CMAKE_CURRENT_SOURCE_DIR}/shared_tests_instance/ie_class/*.cpp
         ${CMAKE_CURRENT_SOURCE_DIR}/shared_tests_instance/single_layer_tests/*.cpp)
 
-if(USE_CNNNETWORK_LPT)
-    file(GLOB CLDNN_TEST_SOURCES
-        ${CLDNN_TEST_SOURCES}
-        ${CMAKE_CURRENT_SOURCE_DIR}/shared_tests_instance/transformations/*.cpp)
```
Comment (ilya-lavrenov) [inline inference-engine/tests_deprecated/functional/cldnn/CMakeLists.txt:23]: who will remove all files with tests?

File: inference-engine/src/mkldnn_plugin/CMakeLists.txt [hunk 2]
```diff
@@ -173,11 +173,7 @@ target_compile_definitions(${TARGET_NAME} PUBLIC -DMKLDNN_THR=${MKLDNN_THR})
 target_link_libraries(${TARGET_NAME} PRIVATE mkldnn inference_engine inference_engine_legacy
                                              inference_engine_transformations)
 
-if(USE_CNNNETWORK_LPT)
-    target_link_libraries(${TARGET_NAME} PRIVATE inference_engine_lp_transformations_legacy)
-else()
-    target_link_libraries(${TARGET_NAME} PRIVATE inference_engine_lp_transformations)
-endif()
+target_link_libraries(${TARGET_NAME} PRIVATE inference_engine_lp_transformations)
```
Comment (ilya-lavrenov) [inline inference-engine/src/mkldnn_plugin/CMakeLists.txt:176]: let's merge with the previous `target_link_libraries`

File: inference-engine/src/mkldnn_plugin/CMakeLists.txt [hunk 3]
```diff
@@ -207,15 +203,8 @@ target_include_directories(${TARGET_NAME}_obj PRIVATE $<TARGET_PROPERTY:inferenc
...
                                                       $<TARGET_PROPERTY:openvino::itt,INTERFACE_INCLUDE_DIRECTORIES>)
 
-if(USE_CNNNETWORK_LPT)
-    target_include_directories(${TARGET_NAME}_obj PRIVATE
-        $<TARGET_PROPERTY:inference_engine_lp_transformations_legacy,INTERFACE_INCLUDE_DIRECTORIES>)
-    target_compile_definitions(${TARGET_NAME}_obj PRIVATE
-        $<TARGET_PROPERTY:inference_engine_lp_transformations_legacy,INTERFACE_COMPILE_DEFINITIONS>)
-else()
-    target_include_directories(${TARGET_NAME}_obj PRIVATE
+target_include_directories(${TARGET_NAME}_obj PRIVATE
```
Comment (ilya-lavrenov) [inline inference-engine/src/mkldnn_plugin/CMakeLists.txt:206]: let's merge with the previous `target_include_directories`

File: inference-engine/tests_deprecated/functional/cldnn/CMakeLists.txt (hunk 1 above)
Comment (eshoguli) [reply inference-engine/tests_deprecated/functional/cldnn/CMakeLists.txt:23]: done, thanks

File: inference-engine/src/mkldnn_plugin/CMakeLists.txt (hunk 2 above)
Comment (eshoguli) [reply inference-engine/src/mkldnn_plugin/CMakeLists.txt]: done, thanks

File: inference-engine/src/mkldnn_plugin/CMakeLists.txt (hunk 3 above)
Comment (eshoguli) [reply inference-engine/src/mkldnn_plugin/CMakeLists.txt]: done, thanks

File: inference-engine/src/cldnn_engine/CMakeLists.txt [hunk 4]
```diff
@@ -25,11 +25,7 @@ target_link_libraries(${TARGET_NAME} PRIVATE clDNN_lib pugixml
                                              inference_engine inference_engine_legacy
                                              inference_engine_transformations)
 
-if(USE_CNNNETWORK_LPT)
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/CMakeLists.txt:28]: What's about `USE_CNNNETWORK_LPT` usages in the plugins code?

Comment (eshoguli) [reply inference-engine/src/cldnn_engine/CMakeLists.txt:28]: done, thanks!

File: inference-engine/tests_deprecated/functional/shared_tests/CMakeLists.txt [hunk 5]
```diff
@@ -13,25 +13,11 @@ list(APPEND SHARED_LIBRARIES
...
-    file(GLOB SHARED_TESTS_SRC
+file(GLOB SHARED_TESTS_SRC
         ${CMAKE_CURRENT_SOURCE_DIR}/common_single_layer_tests/*.cpp
         ${CMAKE_CURRENT_SOURCE_DIR}/lstm/*.cpp
         ${CMAKE_CURRENT_SOURCE_DIR}/graph_tools/*.cpp
-        # requires legacy LPT
-        ${CMAKE_CURRENT_SOURCE_DIR}/network_tests/*.cpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/transformations/*.cpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/transformations/*.hpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/transformations/common/*.cpp
```
Comment (ilya-lavrenov) [inline inference-engine/tests_deprecated/functional/shared_tests/CMakeLists.txt:25]: what about these files? 
```
        ${CMAKE_CURRENT_SOURCE_DIR}/network_tests/*.cpp
        ${CMAKE_CURRENT_SOURCE_DIR}/transformations/*.cpp
        ${CMAKE_CURRENT_SOURCE_DIR}/transformations/*.hpp
        ${CMAKE_CURRENT_SOURCE_DIR}/transformations/common/*.cpp
```
And the same for MKLDNN plugin tests

Comment (eshoguli) [reply inference-engine/tests_deprecated/functional/shared_tests/CMakeLists.txt:25]: done, thanks

**Review by ilya-lavrenov — APPROVED** [review]

**Review by vladimir-paramuzov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
@eshoguli  Seems like the build still fails

**Pull request merged**